cmd/fs-v1.go

```diff
@@ -255,24 +255,18 @@ func (fs fsObjects) ListBuckets() ([]BucketInfo, error) {
 		if isReservedOrInvalidBucket(entry) {
 			continue
 		}
-		var fi os.FileInfo
-		fi, err = fsStatDir(pathJoin(fs.fsPath, entry))
-		if err != nil {
-			// If the directory does not exist, skip the entry.
-			if errorCause(err) == errVolumeNotFound {
-				continue
-			} else if errorCause(err) == errVolumeAccessDenied {
-				// Skip the entry if its a file.
-				continue
-			}
-			return nil, err
+		fi, _ := fsStatDir(pathJoin(fs.fsPath, entry))
+		// There seems like no practical reason to check for errors
+		// at this point, if there are indeed errors we can simply
+		// just ignore such buckets and list only those which
+		// return proper Stat information instead.
+		if fi != nil {
```

Comment:
checking `err` is appropriate way than `fi` i.e. `err == nil`.  If `err` is not nil, we don't even consider what `fi` has.